x-pack/plugins/security/server/authentication/providers/saml.ts

```diff
@@ -341,10 +330,7 @@ export class SAMLAuthenticationProvider extends BaseAuthenticationProvider {
       return AuthenticationResult.failed(err);
     }
 
-    if (
-      newUserAuthenticationResult.user.username !== user.username ||
-      newUserAuthenticationResult.user.authentication_realm.name !== user.authentication_realm.name
-    ) {
+    if (newState.username !== existingState.username) {
```

Comment:
If there's a user logged into Kibana in version 7.4, their state won't have a username. Now after the 7.5 upgrade happens, if there's an IdP initiated login they're going to see the overwritten session page. This is a super corner-case...

If we changed this to:
```
if (existingState.username != null && newState.username !== existingState.username) {
```
We could not display the overwritten session page in this situation, but we also lose the warning if the session is changing who the user is logged in as. Perhaps the way you have it right now is safest?

Reply:
Honestly never considered such thing as a problem that is worth maintaining even a tiny BWC fix for :man_shrugging:  These are such corner cases that both options look acceptable to me, but the current one is cheaper since it doesn't require us to have a clarifying comment for `existingState.username != null` that we'll need to remove in 9.0... 

Having said that if you think it's worth fixing I can just keep passing additional `user: AuthenticatedUser` argument to this function that will have a username of the current user (we'll have it anyway since we use existing state to authenticate user even if new SAML response is present) and we won't need to rely on `existingState` here and handle both corner cases. I just removed it in attempt to simplify code:
```ts
if (user.username !== newState.username) {
```